github/Authorization.py

```diff
@@ -27,106 +27,98 @@
 # along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
 #                                                                              #
 ################################################################################
+from datetime import datetime
+from typing import TYPE_CHECKING, List, Optional
 
 import github.AuthorizationApplication
 import github.GithubObject
 
+from .GithubObject import Attribute, NotSet, Opt, _NotSetType
+
+if TYPE_CHECKING:
+    from .AuthorizationApplication import AuthorizationApplication
+
 
 class Authorization(github.GithubObject.CompletableGithubObject):
     """
     This class represents Authorizations. The reference can be found here https://docs.github.com/en/enterprise-server@3.0/rest/reference/oauth-authorizations
     """
 
-    def __repr__(self):
+    _app: Attribute["AuthorizationApplication"]
+    _created_at: Attribute[datetime]
+    _id: Attribute[int]
+    _note: Attribute[Optional[str]]
+    _note_url: Attribute[Optional[str]]
+    _scopes: Attribute[List[str]]
+    _token: Attribute[str]
+    _updated_at: Attribute[datetime]
+    _url: Attribute[str]
+
+    def __repr__(self) -> str:
         return self.get__repr__({"scopes": self._scopes.value})
 
     @property
-    def app(self):
-        """
-        :type: :class:`github.AuthorizationApplication.AuthorizationApplication`
-        """
+    def app(self) -> "AuthorizationApplication":
         self._completeIfNotSet(self._app)
         return self._app.value
 
     @property
-    def created_at(self):
+    def created_at(self) -> datetime:
         """
         :type: datetime.datetime
         """
         self._completeIfNotSet(self._created_at)
         return self._created_at.value
 
     @property
-    def id(self):
-        """
-        :type: integer
-        """
+    def id(self) -> int:
         self._completeIfNotSet(self._id)
         return self._id.value
 
     @property
-    def note(self):
-        """
-        :type: string
-        """
+    def note(self) -> Optional[str]:
         self._completeIfNotSet(self._note)
         return self._note.value
 
     @property
-    def note_url(self):
-        """
-        :type: string
-        """
+    def note_url(self) -> Optional[str]:
         self._completeIfNotSet(self._note_url)
         return self._note_url.value
 
     @property
-    def scopes(self):
-        """
-        :type: list of string
-        """
+    def scopes(self) -> List[str]:
```

Comment:
Actually, `scopes` is a string (comma-separated list of scopes): https://github.com/PyGithub/PyGithub/issues/1943#issuecomment-1582817370
```suggestion
    def scopes(self) -> str:
```
Do you want to fix this here, or in a separate PR?

Reply:
Do you mean the real type of `.scopes` is `str` instead of `list[str]`?

Reply:
Yes, it is a plain string `str`.